core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/HDFSFileDeleteAction.java

```diff
@@ -0,0 +1,115 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.batch.action;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.action.Action;
+import co.cask.cdap.etl.api.action.ActionContext;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/**
+ * Action that deletes file(s) within HDFS in the same cluster.
+ * A user must specify file/directory path.
+ * Optionals include fileRegex
+ */
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name("HDFSFileDeleteAction")
+@Description("Action to delete files on HDFS")
+public class HDFSFileDeleteAction extends Action {
+  private static final Logger LOG = LoggerFactory.getLogger(HDFSFileDeleteAction.class);
+
+  private HDFSDeleteActionConfig config;
+
+  public HDFSFileDeleteAction(HDFSDeleteActionConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void run(ActionContext context) throws Exception {
+    Path path = new Path(config.path);
+
+    FileSystem fileSystem = path.getFileSystem(new Configuration());
+
+    FileStatus[] listFiles;
+    if (config.fileRegex != null) {
+      PathFilter filter = new PathFilter() {
+        private final Pattern pattern = Pattern.compile(config.fileRegex);
+
+        @Override
+        public boolean accept(Path path) {
+          return pattern.matcher(path.getName()).matches();
+        }
+      };
+      listFiles = fileSystem.listStatus(path, filter);
+    } else {
+      listFiles = fileSystem.listStatus(path);
+    }
+
+    for (FileStatus file: listFiles) {
+      path = file.getPath();
+      if (!fileSystem.delete(path, true)) {
+        if (!config.continueOnError) {
+          throw new IOException(String.format("Removal of %s was unsuccessful.", path.toString()));
+        }
+        LOG.warn("Removal of {} was unsuccessful.", path.toString());
+      }
+    }
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+
+  }
+
+  /**
+   * Config class that contains all properties necessary to execute an HDFS delete command.
+   */
+  public class HDFSDeleteActionConfig extends PluginConfig {
+    @Description("The full HDFS path of the file or files that need to be deleted." +
+      " For example: hdfs://hostname/tmp")
```

Comment:
description should include detail about what happens when its a file versus a directory. I would expect this to behave like `hadoop fs -rm -r path`. If the path is a file, its deleted. If the path is a directory and there is no regex, the directory and its contents are deleted. If the path is a directory and there is a regex, only the files and directories matching that regex are deleted. (This is not the current behavior based on the code).